spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -652,6 +679,8 @@ private void waitForConsumersToStart(Set<AsyncMessageProcessingConsumer> process
 
 	@Override
 	protected void shutdownAndWaitOrCallback(@Nullable Runnable callback) {
+		waitForPendingReplies();
```

Comment:
I don't think this is a right place for such a logic.
I believe it has to be a part of that `awaitShutdown` in the end of this `shutdownAndWaitOrCallback()` method.